src/views/board/new-contact-group-board.tsx

```diff
@@ -25,7 +28,19 @@ const NewContactGroupBoard = (): React.JSX.Element => {
 	const onSave = useCallback(() => {
 		client
 			.createContactGroup(nameValue, memberListEmails)
-			.then(() => {
+			.then((contactGroup) => {
+				if (window.location.pathname.includes(ROUTES_INTERNAL_PARAMS.route.contactGroups)) {
+					useContactGroupStore.getState().addContactGroupInSortedPosition(contactGroup);
+					const element = window.document.getElementById(contactGroup.id);
+					if (element) {
+						element.scrollIntoView({ block: 'end' });
+					}
+					context?.navigateTo(`${ROUTES_INTERNAL_PARAMS.route.contactGroups}/${contactGroup.id}`);
+				} else {
+					useContactGroupStore.getState().emptyContactGroups();
+					useContactGroupStore.getState().setOffset(0);
```

Comment:
I'd prefer using a reset function of the store which reset the state to the initial one, without having to reset all internal objects one by one. Consider this patter https://docs.pmnd.rs/zustand/guides/how-to-reset-state